tests/test_ata_core_loader_concurrency.py

```
@@ -0,0 +1,67 @@
+
+import fast_mlsirm
+import fast_mlsirm._ata_core_loader as loader
+
+
+def test_ata_core_does_not_return_partially_initialized_module(monkeypatch):
+    """Concurrent callers must wait until the secondary module finishes initialization."""
+    target = loader._MODULE_NAME
+    sys.modules.pop(target, None)
+
+    fake_core = ModuleType("fast_mlsirm._core")
+    fake_core.__file__ = "/tmp/fake-fast-mlsirm-core.so"
```

Ruff (0.16.1)

[error] Probable insecure usage of temporary file or directory: "/tmp/fake-fast-mlsirm-core.so"

(S108)